x-pack/plugins/security_solution/public/common/components/event_details/cti_details/threat_details_view.tsx

```diff
@@ -125,31 +107,108 @@ const buildThreatDetailsItems = (enrichment: CtiEnrichment) =>
       };
     });
 
+const renderAccordion = (
+  enrichment: CtiEnrichment,
+  index: number,
+  enrichments: CtiEnrichment[]
+) => {
+  const { id, field, provider, type, value } = getEnrichmentIdentifiers(enrichment);
+  const accordionTitle = `${field} ${value} ${provider ? i18n.PROVIDER_PREPOSITION : ''} ${
+    provider ? provider : ''
+  }`;
+
+  return (
+    <StyledEuiAccordion
+      id={id!}
+      key={id}
+      initialIsOpen={true}
+      arrowDisplay={'right'}
+      buttonContent={
+        <OverflowParent>
+          <OverflowContainer>{accordionTitle}</OverflowContainer>
+        </OverflowParent>
+      }
+      extraAction={
+        isInvestigationTimeEnrichment(type) ? (
+          <EuiFlexItem grow={false}>
+            <InspectButton queryId={QUERY_ID} title={i18n.INVESTIGATION_QUERY_TITLE} />
+          </EuiFlexItem>
+        ) : (
+          <EuiFlexItem />
+        )
+      }
+    >
+      <StyledEuiInMemoryTable
+        columns={columns}
+        compressed
+        data-test-subj={`threat-details-view-${index}`}
+        items={buildThreatDetailsItems(enrichment)}
+      />
+      {index < enrichments.length - 1 && <EuiSpacer size="m" />}
+    </StyledEuiAccordion>
+  );
+};
+
+const renderSortedEnrichments = (enrichments: CtiEnrichment[]) =>
+  enrichments.sort((a, b) => getFirstSeen(b) - getFirstSeen(a)).map(renderAccordion);
+
 const ThreatDetailsViewComponent: React.FC<{
   enrichments: CtiEnrichment[];
 }> = ({ enrichments }) => {
-  const sortedEnrichments = enrichments.sort((a, b) => getFirstSeen(b) - getFirstSeen(a));
+  const {
+    indicator_match_rule: indicatorMatches,
```

Comment:
We're now assuming there will be these values, and only these values for `matched.type`. We should capture these values in some kind of constant, and decide what to do if we find values outside of the expected ones.

Reply:
updated, also handling the case when there is no `matched.type`